src/meta-srv/src/region/supervisor.rs

```diff
@@ -688,56 +698,110 @@ impl RegionSupervisor {
         Ok(tasks)
     }
 
-    async fn do_failover(&mut self, task: RegionMigrationProcedureTask, count: u32) -> Result<()> {
-        let from_peer_id = task.from_peer.id;
-        let to_peer_id = task.to_peer.id;
-        let region_id = task.region_id;
-
-        info!(
-            "Failover for region: {}, from_peer: {}, to_peer: {}, timeout: {:?}, tries: {}",
-            task.region_id, task.from_peer, task.to_peer, task.timeout, count
-        );
-
-        if let Err(err) = self.region_migration_manager.submit_procedure(task).await {
-            return match err {
-                RegionMigrated { .. } => {
-                    info!(
-                        "Region has been migrated to target peer: {}, removed failover detector for region: {}, datanode: {}",
-                        to_peer_id, region_id, from_peer_id
-                    );
-                    self.deregister_failure_detectors(vec![(from_peer_id, region_id)])
-                        .await;
-                    Ok(())
-                }
-                // Returns Ok if it's running or table is dropped.
-                MigrationRunning { .. } => {
-                    info!(
-                        "Another region migration is running, skip failover for region: {}, datanode: {}",
-                        region_id, from_peer_id
-                    );
-                    Ok(())
-                }
-                TableRouteNotFound { .. } => {
-                    self.deregister_failure_detectors(vec![(from_peer_id, region_id)])
-                        .await;
-                    info!(
-                        "Table route is not found, the table is dropped, removed failover detector for region: {}, datanode: {}",
-                        region_id, from_peer_id
-                    );
-                    Ok(())
-                }
-                LeaderPeerChanged { .. } => {
-                    self.deregister_failure_detectors(vec![(from_peer_id, region_id)])
-                        .await;
-                    info!(
-                        "Region's leader peer changed, removed failover detector for region: {}, datanode: {}",
-                        region_id, from_peer_id
-                    );
-                    Ok(())
-                }
-                err => Err(err),
-            };
+    async fn do_failover_tasks(
+        &mut self,
+        tasks: Vec<(RegionMigrationProcedureTask, u32)>,
```

Comment:
Add a comment explaining the purpose of the argument with u32 type.

Reply:
Consider using this structure to replace `Vec<(RegionMigrationProcedureTask, u32)>`. This structure looks clearer.
```
struct RegionMigrationTaskBatch {
    from_peer: Peer,
    to_peer: Peer,
    regions: Vec<(RegionId, u32)>,  // (region_id, failover_count)
}
```